### PR TITLE
feat/add-environment-branching

### PR DESCRIPTION
**Pull request by phoebeschmidt**
Add the ability to supply a `--source` or `--src` param with `space environment create`. 

Currently, this PR includes all changes necessary to make this feature work EXCEPT that this currently relies on a version of the CMA SDK that doesnt do anything with the `source` parameter.

TODO:

1. merge https://github.com/contentful/contentful-management.js/pull/208 
2. release new beta of cma sdk
3. update dependency in this PR

I left off integration test because that just becomes testing if the SDK executes the command correctly with the source param. Am happy to add one if we decide it's worth it, once the new beta has been released.

### CODE REVIEW

File: lib/cmds/space_cmds/environment_cmds/create.js [hunk 1]
```diff
@@ -61,13 +66,16 @@ export async function environmentCreate (argv) {
   const space = await client.getSpace(spaceId)
 
   const options = {}
+  const source = []
```
[the remark this answers is not in the document]
Comment (phoebeschmidt) [reply lib/cmds/space_cmds/environment_cmds/create.js]: so I can spread it in the function call on line 76 and provide no source if the arg isnt provided.

thought it was clever 😂 but happy to change line 76 to a simple `if` if you think that reads better?

[the remark this answers is not in the document]
Comment (Khaledgarbaya) [reply lib/cmds/space_cmds/environment_cmds/create.js]: I think an `if` statement would read better for sure

File: test/integration/cmds/space/environment/__snapshots__/create.test.js.snap [hunk 2]
```diff
@@ -29,6 +31,8 @@ Options:
   -h, --help              Show help                                    [boolean]
   --environment-id, -e    Id of the environment to create             [required]
   --name, -n              Name of the environment to create           [required]
+  --source, --src         ID of the source environment to create the new
```
Comment (Khaledgarbaya) [inline test/integration/cmds/space/environment/__snapshots__/create.test.js.snap]:
```suggestion
--source, -src         ID of the source environment to create the new
```

**Pull request merged**